awx/main/access.py

```diff
@@ -588,17 +588,41 @@ def can_delete(self, obj):
 
 
 class InstanceGroupAccess(BaseAccess):
+    """
+    I can see Instance Groups when I am:
+       - a superuser(system administrator)
+       - an organization admin
+       - at least read_role on the instance group
+    I can edit Instance Groups when I am:
+       - a superuser
+       - an organization admin
+       - admin role on the Instance group
+    I can add/delete Instance Groups:
+       - a superuser(system administrator)
+    """
+
     model = InstanceGroup
     prefetch_related = ('instances',)
 
     def filtered_queryset(self):
-        return InstanceGroup.objects.filter(organization__in=Organization.accessible_pk_qs(self.user, 'admin_role')).distinct()
+        if self.user.admin_of_organizations.exists():
+            return InstanceGroup.objects.filter(organization__in=Organization.accessible_pk_qs(self.user, 'admin_role')).distinct()
```

Comment:
This `return` would not respect the case that a user is an org admin, but has been granted access to an instance group which is not associated with their organization. They would see IGs associated with their org, but not those they got explicit access to.

I'm not saying I _want_ them to, because this would complicate querysets significantly, and that's really something we would like to avoid.

Reply:
I think I see what you're saying, I'm tempted to add a not statement to make it work like you'd expect

Reply:
With the migration logic you added, I think this isn't needed anymore?